pcdsdevices/pseudopos.py

```diff
@@ -0,0 +1,83 @@
+from ophyd.device import Component as Cpt
+from ophyd.pseudopos import PseudoPositioner, PseudoSingle
+
+
+class SyncAxesBase(PseudoPositioner):
+    """
+    Synchronized Axes.
+
+    This will move all axes in a coordinated way, retaining offsets.
+
+    This can be configured to report its position as the min, max, mean, or any
+    custom function acting on a list of positions. Min is the default.
+
+    You should subclass this by adding real motors as components. The class
+    will pick them up and include them correctly into the coordinated move.
+
+    Parameters
+    ----------
+    position_mode: ``func``, optional
+        The function to apply to the list of current positions to determine
+        the combined position. By default, this is the minimum of all axis
+        positions.
+
+    Attributes
+    ----------
+    axes: ``list of str``
+        Names of the components that correspond to the real axes.
+    """
+    pseudo = Cpt(PseudoSingle)
+
+    _default_mode = min
+
+    def __init__(self, *args, position_mode=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._mode = position_mode or self._default_mode
+        self._offsets = {}
+
+    def save_offsets(self, **offsets):
+        """
+        Save the current offsets for the synchronized assembly.
+
+        If not done earlier, this will be automatically run before it is first
+        needed (generally, right before the first move).
+
+        Note that this method does not protect you from getting your offsets
+        into an inconsistent state! Pass arguments into the `offsets` param at
+        your own risk.
+
+        Parameters
+        ----------
+        **offsets: ``int`` or ``float``, optional
+            Mapping of axis name to offset. If omitted, we'll use the current
+            positions. If a partial dictionary is provided, the omitted values
+            are assumed to be either 0 (if no value is available) or unchanged
+            (if previous saved values are available).
+        """
```

Comment:
Maybe a `logger.info` message in here. The operator should be informed when we update the `offsets`

Reply:
Sure, sounds reasonable.

Reply:
On further consideration, is this needed? It will happen every first time you move a SyncAxes... e.g. `ccm.insert()` will start with `INFO: Offsets saved`

Reply:
hmmm. well a `debug` at the least.